pyanaconda/ui/lib/storage.py

```diff
@@ -219,3 +223,44 @@ def try_populate_devicetree():
                 continue
         else:
             break
+
+
+def apply_partitioning(partitioning, show_message):
+    """Apply the given partitioning.
+
+    :param partitioning: a DBus proxy of a partitioning
+    :param show_message: a callback for showing a message
+    :return: an instance of ValidationReport
+    """
+    report = ValidationReport()
+
+    try:
+        show_message(_("Saving storage configuration..."))
+        task_path = partitioning.ConfigureWithTask()
+        task_proxy = STORAGE.get_proxy(task_path)
+        sync_run_task(task_proxy)
+    except StorageConfigurationError as e:
+        show_message(_("Failed to save storage configuration"))
+        report.error_messages.append(str(e))
+        reset_bootloader()
+        reset_storage(scan_all=True)
+    except BootloaderConfigurationError as e:
+        show_message(_("Failed to save boot loader configuration"))
+        report.error_messages.append(str(e))
+        reset_bootloader()
+    else:
+        show_message(_("Checking storage configuration..."))
+        task_path = partitioning.ValidateWithTask()
+        task_proxy = STORAGE.get_proxy(task_path)
+        sync_run_task(task_proxy)
+        report = ValidationReport.from_structure(
+            task_proxy.GetResult()
+        )
+
+    if report.is_valid():
```

Comment:
Is moving out of else clause here intentional ? Empty report is_valid.

Reply:
Fixed.